x/application/module/abci.go

```diff
@@ -20,5 +20,9 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
 		return err
 	}
 
+	if err := k.EndBlockerTransferApplication(ctx); err != nil {
```

Comment:
1. Do we know whether Tokenomics end blocker runs first? We need to settle the session before transferring.
2. Need to find a way to enforce & document this (in the code)

Reply:
Great question! :thinking: 

_TL;DR, application module end blockers currently run first; we would need to change the order of the module names in [`endBlockers`](https://github.com/pokt-network/poktroll/blob/main/app/app_config.go#L178). Just need to make sure that doesn't cause any other issue(s)._

[Cosmos-sdk docs](https://docs.cosmos.network/main/build/building-modules/beginblock-endblock) say:
> It is possible for developers to define the order of execution between the BeginBlocker/EndBlocker functions of each of their application's modules via the module's manager SetOrderBeginBlocker/SetOrderEndBlocker methods.

We have an `endBlockers` var defined in [`app_config.go:178`](https://github.com/pokt-network/poktroll/blob/main/app/app_config.go#L178):
![image](https://github.com/user-attachments/assets/f8e244da-ca7c-4baf-a326-a212a0a0f775)

Which is passed into a `ModuleConfig` [below](https://github.com/pokt-network/poktroll/blob/main/app/app_config.go#L249) that is passed to `ModuleManager.SetOrderEndBlockers()` [in the cosmos-sdk runtime by `App#Load()`](https://github.com/cosmos/cosmos-sdk/blob/v0.50.7/runtime/app.go#L131)

We could enforce this with a script that runs in CI that knows about the constraints but it may be just as effective to leave comments in/around the `endBlockers` assignment articulating the constraints.